config/config.go

```diff
@@ -0,0 +1,176 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/pflag"
+	c "nav/constants"
+)
+
+type Config struct {
+	ConfValues ConfValues
+	// add DB connection instance here
+}
+
+type ConfValues struct {
+	Symbol         string    `json:"symbol"`
+	Type           string    `json:"output_type"`
+	DBDriver       string    `json:"db_driver"`
+	DBDSN          string    `json:"DBDSN"`
+	ExcludedBefore []string  `json:"excluded_before"`
+	ExcludedAfter  []string  `json:"excluded_after"`
+	TargetSubsys   []string  `json:"target_subsys"`
+	MaxDepth       int       `json:"max_depth"`
+	Mode           c.OutMode `json:"mode"`
+	DBInstance     int       `json:"db_instance"`
+}
+
+// New creates a new Config instance and returns a pointer to it.
+func New() (*Config, error) {
+	config := Config{}
+	confValues, err := initConfig()
+	config.ConfValues = confValues
+	if err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+// initConfig gathers the configuration values from command line and config file (if provided) and returns
+// them in a ConfValues struct. If both are provided, the command line flags take precedence.
+func initConfig() (ConfValues, error) {
+	fs := pflag.NewFlagSet("nav", pflag.ContinueOnError)
+	var confValues ConfValues
+	var configPath string
+
+	err := parseCommandLine(fs, &configPath, os.Args)
+	if err != nil {
+		return ConfValues{}, fmt.Errorf("error initializing command line flags: %w", err)
+	}
+
+	if len(os.Args) == 1 {
```

Comment:
I'd move this check as a guard clause to the top. Nothing to parse in this case.

Reply:
My initial idea for configuration was to include a default config, the ability to define items using a configuration file, and to accept command line arguments. According to this approach, the working configuration should have been the combination of all three in the order listed, with the default being overwritten by the configuration file and the sum of the two being overwritten by command line arguments, making the latter more powerful than the former two. Admittedly, my own laziness often led me to rely on the default configuration, which was sufficient in most cases. While I understand that having a default for only one user is not necessary, I believe that other users with similar tendencies might appreciate such a feature. Since we cannot provide a custom binary, perhaps we could define a default configuration file name that is used when the binary is called without arguments.

Reply:
@alessandrocarminati Maybe this is something that could be written down as a Github issue and addressed in some future PR?

Reply:
https://github.com/AC-FuSa-Tools/nav/issues/26